### PR TITLE
correctly merge runfiles in py_appengine_binary. fixes #44

### LINKED ISSUE #44: py_appengine_binary doesn't work properly
There multiple issues with that rule:
1. It doesn't maintain creating of __init__.py files throughout paths.
2. External modules don't get imported if they import their submodules inside.
Actually py_binary() rule does all of that but it's not aware of .yaml configs.

### PR DESCRIPTION
**Pull request by maximermilov**

### CODE REVIEW

File: appengine/py_appengine.bzl [hunk 1]
```diff
@@ -155,10 +156,12 @@ def py_appengine_binary(name, srcs, configs, deps=[], data=[]):
   """Convenience macro that builds the app and offers an executable
      target to deploy on Google app engine.
   """
-  native.py_library(
+  # uses py_binary because it generates __init__.py files
+  native.py_binary(
       name="_py_appengine_" + name,
       srcs = srcs,
       deps=deps,
+      main=srcs[0],  # no entry point, use arbitrary source file
```
Comment (kchodorow) [inline appengine/py_appengine.bzl]: Check that len(srcs) > 0.

File: appengine/py_appengine.bzl [hunk 2]
```diff
@@ -92,12 +92,13 @@ def _py_appengine_binary_base_impl(ctx):
     files = c.files.to_list()
     for f in files:
       symlinks[f.basename] = f
+
   runfiles = ctx.runfiles(
-      transitive_files=ctx.attr.binary.data_runfiles.files
-      + ctx.attr.devappserver.data_runfiles.files
+      transitive_files=ctx.attr.devappserver.data_runfiles.files
       + ctx.attr.appcfg.data_runfiles.files,
```
Comment (kchodorow) [inline appengine/py_appengine.bzl]: Should this also be merged?

File: appengine/py_appengine.bzl (hunk 1 above)
Comment (maximermilov) [inline appengine/py_appengine.bzl]: Done

File: appengine/py_appengine.bzl (hunk 2 above)
Comment (maximermilov) [inline appengine/py_appengine.bzl]: Done

**Pull request merged**